internal/translation/audit/conversion_test.go

```diff
@@ -0,0 +1,220 @@
+package audit
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.mongodb.org/atlas-sdk/v20231115008/admin"
+
+	akov2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1"
+)
+
+func TestConversionContructor(t *testing.T) {
+	testCases := []struct {
+		title          string
+		input          *akov2.Auditing
+		expectedOutput *AuditConfig
+	}{
+		{
+			title: "Just enabled",
+			input: &akov2.Auditing{
+				Enabled: true,
+			},
+			expectedOutput: &AuditConfig{
+				&akov2.Auditing{
+					Enabled:     true,
+					AuditFilter: `{}`,
+				},
+			},
+		},
+		{
+			title: "Auth success logs as well",
+			input: &akov2.Auditing{
+				Enabled:                   true,
+				AuditAuthorizationSuccess: true,
+			},
+			expectedOutput: &AuditConfig{
+				&akov2.Auditing{
+					Enabled:                   true,
+					AuditAuthorizationSuccess: true,
+					AuditFilter:               `{}`,
+				},
+			},
+		},
+		{
+			title: "With a filter",
+			input: &akov2.Auditing{
+				Enabled:     true,
+				AuditFilter: `{"atype":"authenticate"}`,
+			},
+			expectedOutput: &AuditConfig{
+				&akov2.Auditing{
+					Enabled:     true,
+					AuditFilter: `{"atype":"authenticate"}`,
+				},
+			},
+		},
+		{
+			title: "With a filter and success logs",
+			input: &akov2.Auditing{
+				Enabled:                   true,
+				AuditAuthorizationSuccess: true,
+				AuditFilter:               `{"atype":"authenticate"}`,
+			},
+			expectedOutput: &AuditConfig{
+				&akov2.Auditing{
+					Enabled:                   true,
+					AuditAuthorizationSuccess: true,
+					AuditFilter:               `{"atype":"authenticate"}`,
+				},
+			},
+		},
+		{
+			title: "All set but disabled",
+			input: &akov2.Auditing{
+				AuditAuthorizationSuccess: true,
+				AuditFilter:               `{"atype":"authenticate"}`,
+			},
+			expectedOutput: &AuditConfig{
+				&akov2.Auditing{
+					AuditAuthorizationSuccess: true,
+					AuditFilter:               `{"atype":"authenticate"}`,
+				},
+			},
+		},
+		{
+			title: "Default (disabled) case",
+			input: &akov2.Auditing{},
+			expectedOutput: &AuditConfig{
+				&akov2.Auditing{
+					AuditFilter: `{}`,
+				},
+			},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.title, func(t *testing.T) {
+			actualResult := NewAuditConfig(tc.input)
+			assert.Equal(t, tc.expectedOutput, actualResult)
+		})
+	}
+}
+
+func TestToAndFromAtlas(t *testing.T) {
```

Comment:
This seems like a weird mix, I would just have a `TestToAtlas` and a separate `TestFromAtlas` with only internal types or atlas types respectively.

Alternatively one thing you could do is to have a single `TestToAndFromAtlas` table with only the internal type as inputs, then in the test you do `ToAtlas` and the result you pass to `FromAtlas` and finally compare the final result with the original input, which should match.